diff --git a/comp2.c b/comp2.c
index 66a323a..6abaec0 100644
--- a/comp2.c
+++ b/comp2.c
@@ -39,9 +39,9 @@ main(void)
 	sigprocmask(SIG_UNBLOCK, &set, NULL);
 
 	while (1) {
-		fprintf(stderr,
-		        "wasting one file descriptor every 5s... :(\n");
-		open("./file", O_RDONLY | O_CREAT, S_IRUSR | S_IWUSR);
+		fprintf(stderr, "no more resource leaks! :)\n");
+		int fd = open("./file", O_RDONLY | O_CREAT, S_IRUSR | S_IWUSR);
+		close(fd);
 		sleep(5);
 	}
 }
